haystack/pipelines/utils.py

```diff
@@ -297,18 +305,22 @@ def _format_wrong_examples_report(
         "document_id", "context", "document_id_and_context", "document_id_or_context", "answer", "document_id_or_answer"
     ] = "document_id_or_answer",
     answer_scope: Literal["any", "context", "document_id", "document_id_and_context"] = "any",
+    wrong_examples_filter: List[str] = None,
+    max_characters_per_wrong_examples_report: int = None,
 ):
     examples = {
         node: eval_result.wrong_examples(
             node, document_scope=document_scope, answer_scope=answer_scope, n=n_wrong_examples
         )
         for node in eval_result.node_results.keys()
     }
-    examples_formatted = {
-        node: "\n".join(map(_format_wrong_example, examples)) for node, examples in examples.items() if any(examples)
-    }
+    examples_formatted = {}
+    for node, examples in examples.items():  # type: ignore
+        if any(examples):
+            examples_formatted[node] = "\n".join([_format_wrong_example(e, wrong_examples_filter) for e in examples])  # type: ignore
 
-    return "\n".join(map(_format_wrong_examples_node, examples_formatted.keys(), examples_formatted.values()))
+    final_result = "\n".join(map(_format_wrong_examples_node, examples_formatted.keys(), examples_formatted.values()))
+    return final_result[:max_characters_per_wrong_examples_report]
```

Comment:
I think it would be nicer to have this limit on the property level of each example.

Reply:
Ok, but we have the field filter for that. This is more like brute force. Cut the report to this many characters - I don't care type of a thing.

Reply:
No, I mean that you cut e.g. the content field of documents. This might be pretty long compared to all the other fields and it should not suppress all the other fields.